### PR TITLE
fix(auth): Hide splash screen on android SDK lower than 28 for biometrics

### PR DESCRIPTION
**Pull request by michalszorad**
Resolves [YOMO-682](https://emurgo.atlassian.net/jira/software/c/projects/YOMO/issues/YOMO-682)

[YOMO-682]: https://emurgo.atlassian.net/browse/YOMO-682?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

On SDKs lower than 28, the fingerprint layer is covered by the splash screen, so we need to close the splash screen to allow the user to authenticate.

https://github.com/Emurgo/yoroi/assets/19684576/79510091-2f5a-4905-b4f6-02d677dfee84

### CODE REVIEW

File: apps/wallet-mobile/src/AppNavigator.tsx [hunk 1]
```diff
@@ -37,6 +38,18 @@ export const AppNavigator = () => {
 
     // try first OS auth before navigating to os login screen
     if (authAction === 'auth-with-os') {
+      if (Platform.OS === 'android') {
+        supportsAndroidFingerprintOverlay().then((isOverlaySupported) => {
+          if (!isOverlaySupported) {
+            RNBootSplash.hide({fade: true})
```
Comment (michalszorad) [inline apps/wallet-mobile/src/AppNavigator.tsx:44]: On SDKs lower than 28, we need to hide the splash screen, so the user is able to authenticate.

File: apps/wallet-mobile/src/auth/OsAuthScreen/OsAuthScreen.tsx [hunk 2]
```diff
@@ -31,9 +31,14 @@ export const OsAuthScreen = ({
   const [showImage, setShowImage] = React.useState(showFingerPlaceholder)
 
   React.useEffect(() => {
-    DeviceInfo.getApiLevel().then((sdk) => {
-      setShowImage((Platform.OS === 'android' && sdk < 28) || showFingerPlaceholder)
-    })
+    if (Platform.OS === 'android') {
+      supportsAndroidFingerprintOverlay().then((isSupported) => {
```
Comment (michalszorad) [inline apps/wallet-mobile/src/auth/OsAuthScreen/OsAuthScreen.tsx:35]: Refactored the check here

File: apps/wallet-mobile/src/auth/OsAuthScreen/OsAuthScreen.tsx [hunk 3]
```diff
@@ -31,9 +31,14 @@ export const OsAuthScreen = ({
   const [showImage, setShowImage] = React.useState(showFingerPlaceholder)
 
   React.useEffect(() => {
-    DeviceInfo.getApiLevel().then((sdk) => {
-      setShowImage((Platform.OS === 'android' && sdk < 28) || showFingerPlaceholder)
-    })
+    if (Platform.OS === 'android') {
```
Comment (stackchain) [inline apps/wallet-mobile/src/auth/OsAuthScreen/OsAuthScreen.tsx:34]: Platform is checked twice.

Comment (michalszorad) [reply apps/wallet-mobile/src/auth/OsAuthScreen/OsAuthScreen.tsx:34]: I've removed the check from `supportsAndroidFingerprintOverlay`

**Review by stackchain — APPROVED** [review]

**Pull request merged**